presc/evaluations/conditional_metric.py

```diff
@@ -0,0 +1,164 @@
+            width_interval = 1
+        else:
+            result_edges = self.bins[:-1]
+            alignment = "edge"
+            # First element will be NaN.
+            bin_widths = self.bins.diff()[1:]
+            width_interval = bin_widths * _config["plot_width_fraction"]
+
+        plt.ylim(0, 1)
+        plt.xlabel(xlab)
+        plt.ylabel(ylab)
+        plt.bar(
+            result_edges,
+            self.vals,
+            width=width_interval,
+            bottom=None,
+            align=alignment,
+            edgecolor="white",
+            linewidth=2,
+        )
+        plt.show(block=False)
+
+
+class ConditionalMetric:
+    """Computation of confusion-based metrics across subsets of a test dataset.
+
+    model: the ClassificationModel to run the evaluation for
+    test_dataset: a Dataset to use for evaluation.
+    config: optional dict of config options to override the defaults
+    """
+
+    def __init__(self, model, test_dataset, config=None):
+        _config = dict(DEFAULT_CONFIG)
+        if config:
+            _config.update(config)
+        self._config = _config
+        self._model = model
+        self._test_dataset = test_dataset
+        self._test_pred = self._model.predict_labels(test_dataset)
+
+    def compute_for_column(self, colname, metric, as_categorical=False):
+        """Compute the evaluation for the given dataset column.
+
+        The metric is computed within unique values of the specified column
+        (if categorical) or within bins partitioning its range (if continuous).
+
+        colname: a column in the dataset to partition on
+        metric: the evaluation metric to compute across the partitions. This should be
+            a function f(y_true, y_pred) which accepts Series of true and
+            predicted labels.
+        as_categorical: should the feature be treated as categorical, ie. binned
+            on its unique values? If the feature is not numeric, this param is
+            ignored.
+
+        Returns a `ConditionalMetricResult` instance.
+        """
+        return compute_conditional_metric(
+            grouping_col=self._test_dataset.df[colname],
+            true_labs=self._test_dataset.labels,
+            pred_labs=self._test_pred,
+            metric=metric,
+            as_categorical=as_categorical,
+            config=self._config,
+        )
+
+    def display(self, colnames=None):
+        """Displays the conditional metric result for each specified column.
+
+        colnames: a list of column names to run the evaluation over, creating a plot
+            for each. If not supplied, defaults to all feature columns.
+        """
+        if colnames is None:
+            colnames = self._test_dataset.feature_names
+        for colname in colnames:
+            # TODO: don't hardcode the metric
+            eval_result = self.compute_for_column(colname, metric=METRIC)
+            eval_result.display_result(xlab=colname, ylab="Metric value")
```

Comment:
I passed this in as a function arg. However this will prob get updated further as I've temporarily fixed the metric but that will get passed in as well.